src/main/java/it/unitn/disi/smatch/data/trees/IBaseNode.java

```diff
@@ -42,6 +42,16 @@
      * @return unmodifiable list of receivers children
      */
     List<E> getChildren();
+    
+    /**
+     * Returns a list of children that can be modified. 
+     * Notice that acting on this list won't fire events.
```

Comment:
Notably, this exception opens "a hole" in the API. Namely, now the tree modification change notification could be broken, because event can't be triggered.

You would be better off not doing this change, but I don't know what your needs are now and can't give you an elegant solution.